src/Workspaces/Core/Portable/Editing/SyntaxEditor.cs

```diff
@@ -40,7 +43,7 @@ internal SyntaxEditor(SyntaxNode root, SyntaxGenerator generator)
             _changes = new List<Change>();
         }
 
-        private SyntaxNode ApplyTrackingToNewNode(SyntaxNode node)
+        private SyntaxNode? ApplyTrackingToNewNode(SyntaxNode? node)
```

Comment:
Does this only return null because of the guard condition for `node` being null? If so, a `[NotNullIfNotNull]` could help here.